### ISSUE TITLE
QueryCategory property allows over 30 chars but this will fail in the cloud with an error...

### ISSUE THREAD

**Issue by eque2al** [microsoft/AL#6418]
**Describe the bug**
The QueryCategory property using in Pages is permitting over 30 characters.  However, when you attempt to run the page (in the cloud at least) it fails spectacularly!

**To Reproduce**
Add a QueryCategory over 30 chars to an existing page.  Compile and run it (well, attempt to).

``` AL Code to reproduce the issue ```

**Expected behavior**

**Screenshots**

![image](https://user-images.githubusercontent.com/57717503/104612537-37fb0d80-567e-11eb-8474-135e4788ef8e.png)

**5. Versions:**
BC V17 (plus latest VS Code version as of today)

**Comment by dtkb** [comment]
Does "View the last known error" indicate anything about where the overflow is happening?

**Comment by eque2al** [comment]
Nope - it says "No errors have occurred since you last logged in" - but of course they have.

**Comment by atoader** [comment]
Hi! Please take a minute to read the guidelines for this repository https://github.com/microsoft/AL and our collection of frequently asked questions https://github.com/Microsoft/AL/wiki/frequently-Asked-Questions .

As this repository is only for issues related to the AL compiler in latest developer preview environment for Dynamics 365 Business Central. We suggest that you open a support case or file a bug in Collaborate, to ensure that all Business Central users benefit from your catch as soon as possible.

To open a support case, you can:

- Open a support request to CSS through the PartnerSource portal
- Contact your Service Account Manager (SAM) in your local subsidiary to understand what is included in your contract as of support incident and PAH (Partner Advisory Hours). Your SAM might also direct you "step by step" how to open a support request or how to get credentials, if this is the first time for you or your company

If you file the bug in Collaborate, remember to include steps to reproduce the issue, and the Business Central build number and country version you're using.

**Comment by eque2al** [comment]
This is an issue with the AL compiler as it shouldn't allow so many characters as they will ultimately fail.!  Why has this been closed?

**Comment by dtkb** [comment]
The error sounds more platform-y than compiler-y, but I'd agree that if the platform can't support more than a certain length, the compiler shouldn't let that code reach the platform, so I would agree this should stay open until that is fixed. Such a limitation should be enforced by the compiler and documented (not just one or the other!)

**Comment by eque2al** [comment]
Exactly my point - well put @dtkb

**Comment by atoader** [comment]
Good point, both! I was focused on the fact that the platform should handle the longer text. We can of course make this limitation explicit in the compiler.

**Comment by kalberes** [comment]
@atoader 

I do not think this should be built into the binding logic of the compiler validation. The compiler should not hardcode the runtimes old C/SIDE legacy issues. 
It could be an AL cope validation on the other hand.
